src/testdrive/src/action/kinesis/verify.rs

```diff
@@ -0,0 +1,133 @@
+            };
+            let shard_iterator = match state
+                .tokio_runtime
+                .block_on(state.kinesis_client.list_shards(list_shards_input))
+            {
+                Ok(output) => match output.shards.as_deref() {
+                    Some([shard]) => {
+                        let shard_iterator_input = GetShardIteratorInput {
+                            shard_id: shard.shard_id.clone(),
+                            shard_iterator_type: String::from("TRIM_HORIZON"),
+                            starting_sequence_number: None,
+                            stream_name: stream_name.clone(),
+                            timestamp: None,
+                        };
+                        match state.tokio_runtime.block_on(
+                            state
+                                .kinesis_client
+                                .get_shard_iterator(shard_iterator_input),
+                        ) {
+                            Ok(output) => match output.shard_iterator {
+                                Some(iterator) => iterator,
+                                None => {
+                                    return Err(format!(
+                                        "unable to find a shard iterator for Kinesis stream {}",
+                                        &stream_name
+                                    ))
+                                }
+                            },
+                            Err(e) => {
+                                return Err(format!(
+                                    "hit error trying to get Kinesis shard iterator: {}",
+                                    e.to_string()
+                                ))
+                            }
+                        }
+                    }
+                    None | Some(_) => {
+                        return Err(String::from("Kinesis stream must have exactly one shard."))
+                    }
+                },
+                Err(e) => {
+                    return Err(format!(
+                        "hit error trying to get Kinesis shards: {}",
+                        e.to_string()
+                    ))
+                }
+            };
+
+            let get_records_input = GetRecordsInput {
+                limit: None,
+                shard_iterator,
+            };
+            records = match state
+                .tokio_runtime
+                .block_on(state.kinesis_client.get_records(get_records_input))
+            {
+                Ok(output) => output.records,
+                Err(e) => {
+                    return Err(format!(
+                        "hit error getting Kinesis records: {}",
+                        e.to_string()
+                    ))
+                }
+            };
+        }
+
+        for (expected, actual) in self.expected_messages.iter().zip(records.iter_mut()) {
+            let record_string = match from_utf8(actual.data.as_ref()) {
```

Comment:
Convention is to import `std::str` and call the function as `str::from_utf8`, just so it's a bit more clear what function is getting invoked. (Function names in Rust tend to assume you'll be invoking them as `module::func`, and are a little ambiguous otherwise.)